File: PreventAudioFocus/src/main/java/com/programminghoch10/PreventAudioFocus/Hook.java
package com.programminghoch10.PreventAudioFocus;

import android.media.AudioManager;

import de.robv.android.xposed.IXposedHookLoadPackage;
import de.robv.android.xposed.XC_MethodReplacement;
import de.robv.android.xposed.XposedBridge;
import de.robv.android.xposed.XposedHelpers;
import de.robv.android.xposed.callbacks.XC_LoadPackage;

public class Hook implements IXposedHookLoadPackage {
    @Override
    public void handleLoadPackage(XC_LoadPackage.LoadPackageParam lpparam) {
        Class<?> clazz = AudioManager.class;
        if (lpparam.packageName.equals("android")) clazz = XposedHelpers.findClass("com.android.server.audio.MediaFocusControl", lpparam.classLoader);
        XposedBridge.hookAllMethods(clazz, "requestAudioFocus", XC_MethodReplacement.returnConstant(AudioManager.AUDIOFOCUS_REQUEST_GRANTED));
    }
}

File: PreventAudioFocus/src/main/java/com/programminghoch10/PreventAudioFocus/Hook.kt
package com.programminghoch10.PreventAudioFocus

import android.media.AudioManager
import de.robv.android.xposed.IXposedHookLoadPackage
import de.robv.android.xposed.XC_MethodReplacement
import de.robv.android.xposed.XSharedPreferences
import de.robv.android.xposed.XposedBridge
import de.robv.android.xposed.XposedHelpers
import de.robv.android.xposed.callbacks.XC_LoadPackage.LoadPackageParam

class Hook : IXposedHookLoadPackage {
    override fun handleLoadPackage(lpparam: LoadPackageParam) {
        var clazz: Class<*>? = AudioManager::class.java
        if (lpparam.packageName == "android") clazz = XposedHelpers.findClass(
            "com.android.server.audio.MediaFocusControl", lpparam.classLoader
        )
        val sharedPreferences = XSharedPreferences(BuildConfig.APPLICATION_ID, SHARED_PREFERENCES_NAME)
        val constant = ENTRIES.map {
            Triple(
                it.key,
                it.value,
                sharedPreferences.getBoolean(it.key, false),
            )
        }.find { it.third }?.second ?: ENTRIES_DEFAULT
        XposedBridge.hookAllMethods(
            clazz, "requestAudioFocus", XC_MethodReplacement.returnConstant(constant)
        )
    }
}

